### PR TITLE
adding the viewportindex to the setStartLoadHandler

### PR DESCRIPTION
**Pull request by biharck**

### CODE REVIEW

**Comment by biharck** [comment]
This PF depends on https://github.com/cornerstonejs/cornerstoneTools/pull/1004

File: src/CornerstoneViewport/CornerstoneViewport.js [hunk 1]
```diff
@@ -131,11 +131,16 @@ class CornerstoneViewport extends Component {
       viewport: cornerstone.getDefaultViewport(null, undefined)
     };
 
-    const { loadHandlerManager } = cornerstoneTools;
-    loadHandlerManager.setStartLoadHandler(this.startLoadingHandler);
-    loadHandlerManager.setEndLoadHandler(this.doneLoadingHandler);
-
     this.debouncedResize = debounce(() => {
+      const { loadHandlerManager } = cornerstoneTools;
```
Comment (swederik) [inline src/CornerstoneViewport/CornerstoneViewport.js]: Why did you move these inside the resize handler? Put them back where they were, it makes no sense to re-set these every single time the element is resized.

Comment (biharck) [reply src/CornerstoneViewport/CornerstoneViewport.js]: yes, sir. Done

File: src/CornerstoneViewport/CornerstoneViewport.js [hunk 2]
```diff
@@ -131,11 +133,15 @@ class CornerstoneViewport extends Component {
       viewport: cornerstone.getDefaultViewport(null, undefined)
     };
 
-    const { loadHandlerManager } = cornerstoneTools;
-    loadHandlerManager.setStartLoadHandler(this.startLoadingHandler);
-    loadHandlerManager.setEndLoadHandler(this.doneLoadingHandler);
-
     this.debouncedResize = debounce(() => {
+      loadHandlerManager.setStartLoadHandler(
```
Comment (swederik) [inline src/CornerstoneViewport/CornerstoneViewport.js]: You have to move these calls too.

setStartLoadHandler should not be inside debouncedResize, it makes no sense at all.
Debounced resize is called every single time the viewport is resized. It happens 10s-100s of times.
The way you have the code now, every single time the viewport is resized, we will be pointlessly setting the handlers. There is no reason at all why this needs to be done. Move it out of this function.
